api/cpp/include/slint.h

```diff
@@ -421,12 +425,12 @@ class Window
     /// This function allows registering a callback that's invoked when the user tries to close
     /// a window.
     /// The callback has to return a CloseRequestResponse.
-    template<typename F>
+    // clang-format off
+    template<std::invocable F>
+        requires(std::is_convertible_v<std::invoke_result_t<F>, CloseRequestResponse>)
```

Comment:
I made up this concept to check the return value, I hope this is right.  Strangly, i did not found online the best practice to have function with functor using concept that take a given function signature.

Reply:
Wouldn't it work to use the same expression as in the static assert perhaps?

Reply:
Actually, I think the assert was too strict. Don't you think?

Reply:
I don't think so, but I'm apparently missing the use-case. How would a return type look like that's implicitly convertible? A type that has "operator CloseRequestResponse()"?

Reply:
Yes.
Bit this is about the pattern. 
In this case admittedly there is not much that can convert to that enum, but in the interpreter, for the callback that returns a Value, then it would be anything that can convert to a Value such as int, SharedString, Struct, ...

Reply:
Ok, for this case type there's no use-case then. I didn't know that this was about the pattern.

For the general pattern based on the interpreter example:

If the concept required the return type to be `Value`, then I could use this, I suppose:

```cpp
static Value my_callback()
{
    return 42; // implicitly converted to Value
}
```

and with your convertible concept this:

```cpp
static int my_callback()
{
    return 42; // converted on the Slint side
}
```

If my above understanding is correct, then it seems cool that I can use "my_callback", since that could be a function that comes from somewhere totally different.

I guess this is similar to Rust's pattern of an argument being a `impl Into<Foo>`.

👍 for the interpreter use-case.

Reply:
The point is more like the difference with

```c++
foobar.invoke("foo", [] () -> Value {  return 42; });
// vs
foobar.invoke("foo", [] {  return 42; })
```

IMHO both should work.

Reply:
Right, that's also a compelling example. 👍